### PR TITLE
Add carbon metrics to get dashboard support

### LINKED ISSUE #87: Request support for monitor the extenstion
**Description:**
Allow the siddhhi-io-file extension to monitor the details of source, sink and file operations

**Suggested Labels:**
new-feature

### PR DESCRIPTION
**Pull request by sahandilshan**
## Purpose
$subject
 fix #87 

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Related PRs
https://github.com/wso2/carbon-analytics/pull/1849

### CODE REVIEW

File: component/src/main/java/io/siddhi/extension/execution/file/FileCopyExtension.java [hunk 1]
```diff
@@ -137,6 +140,19 @@ protected StateFactory init(AbstractDefinition inputDefinition, ExpressionExecut
             pattern = Pattern.compile(((ConstantExpressionExecutor)
                     attributeExpressionExecutors[2]).getValue().toString());
         }
+        if (MetricsDataHolder.getInstance().getMetricService() != null &&
+                MetricsDataHolder.getInstance().getMetricManagementService().isEnabled()) {
+            try {
+                if (MetricsDataHolder.getInstance().getMetricManagementService().isReporterRunning(
+                        "prometheus")) {
```
Comment (ramindu90) [inline component/src/main/java/io/siddhi/extension/execution/file/FileCopyExtension.java]: Shall we make this a constant? Maybe it should be a constant in the carbon metrics package.

File: component/src/main/java/io/siddhi/extension/execution/file/FileCopyExtension.java [hunk 2]
```diff
@@ -137,6 +140,19 @@ protected StateFactory init(AbstractDefinition inputDefinition, ExpressionExecut
...
+                    String siddhiAppName = siddhiQueryContext.getSiddhiAppContext().getName();
+                    fileCopyMetrics = new FileCopyMetrics(siddhiAppName);
+                }
+            } catch (IllegalArgumentException e) {
+                log.debug("Prometheus reporter is not running. Hence file metrics will not be initialise in "
```
Comment (ramindu90) [inline component/src/main/java/io/siddhi/extension/execution/file/FileCopyExtension.java]: Maybe this should be rephrased as "will not be initialized"

File: component/src/main/java/io/siddhi/extension/io/file/metrics/FileMoveMetrics.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    private String destination;
+    private long time;
+
+    public FileMoveMetrics(String siddhiAppName) {
+        super(siddhiAppName);
+    }
+
+    public void getMoveMetric(int status) {
+        MetricsManagement.getInstance().getMetricService()
+                .gauge(String.format("io.siddhi.SiddhiApps.%s.Siddhi.File.Operations.Move.%s.%s.%s",
```
Comment (ramindu90) [inline component/src/main/java/io/siddhi/extension/io/file/metrics/FileMoveMetrics.java:39]: Cant change to have 1 metric class and pass "io.siddhi.SiddhiApps.%s.Siddhi.File.Operations.Move.%s.%s.%s" when creating and retrieving the metric?

**Review by ramindu90 — CHANGES_REQUESTED** [review]

File: component/src/main/java/io/siddhi/extension/execution/file/FileCopyExtension.java (hunk 1 above)
Comment (sahandilshan) [reply component/src/main/java/io/siddhi/extension/execution/file/FileCopyExtension.java]: This was fixed in 85ee5fb1ca9a98ffc21692e9a468773fc49cacba, There's no constant available in the carbon metrics, Hence introduce as a new constant.

File: component/src/main/java/io/siddhi/extension/execution/file/FileCopyExtension.java (hunk 2 above)
Comment (sahandilshan) [reply component/src/main/java/io/siddhi/extension/execution/file/FileCopyExtension.java]: This was fixed in 85ee5fb1ca9a98ffc21692e9a468773fc49cacba

**Review by ramindu90 — APPROVED** [review]

**Pull request merged**